code/headers/frame_handler.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <array>
+#include <display_ostream.hpp>
+
+#include <battery_frame_action.hpp>
+#include <distance_frame_action.hpp>
+#include <manual_control_frame_action.hpp>
+#include <temperature_frame_action.hpp>
+
+namespace r2d2::robos {
+    class frame_handler_c {
+    private:
+        base_comm_c &comm;
+
+        display_ostream_c power_ostream{
+            comm, claimed_display_cursor::ROBOS_POWER_CURSOR, 0, 0};
+        display_ostream_c distance_ostream{
+            comm, claimed_display_cursor::ROBOS_DISTANCE_CURSOR, 0, 16};
+        display_ostream_c temperature_ostream{
+            comm, claimed_display_cursor::ROBOS_TEMPERATURE_CURSOR, 0, 32};
+
+    public:
+        frame_handler_c(base_comm_c &comm) : comm(comm) {
+        }
+
+        void process(frame_s &frame) {
+            switch (frame.type) {
```

Comment:
Same as above, is it really necessary to hardcode it all?

Reply:
I believe there is only one alternative: adding all handlers to an array and looping thru them
I think this will result in more code per handler, thus making it more error-prone